packages/ethereum-contracts/test/foundry/deployments/ForkSuperTokenFactoryUpgrade.t.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity 0.8.18;
+
+import { console, Test } from "forge-std/Test.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import {
+    IConstantFlowAgreementV1,
+    ConstantFlowAgreementV1,
+    IConstantFlowAgreementHook
+} from "../../../contracts/agreements/ConstantFlowAgreementV1.sol";
+import {
+    IInstantDistributionAgreementV1
+} from "../../../contracts/interfaces/agreements/IInstantDistributionAgreementV1.sol";
+import { IResolver } from "../../../contracts/interfaces/utils/IResolver.sol";
+import {
+    ISuperfluidGovernance
+} from "../../../contracts/interfaces/superfluid/ISuperfluidGovernance.sol";
+import {
+    SuperfluidLoader
+} from "../../../contracts/utils/SuperfluidLoader.sol";
+import {
+    ISuperfluid
+} from "../../../contracts/interfaces/superfluid/ISuperfluid.sol";
+import {
+    IERC20,
+    ISuperToken,
+    SuperToken
+} from "../../../contracts/superfluid/SuperToken.sol";
+import {
+    ISuperTokenFactory,
+    SuperTokenFactory
+} from "../../../contracts/superfluid/SuperTokenFactory.sol";
+import {
+    SuperTokenFactoryUpdateLogicContractsTester
+} from "../../../contracts/mocks/SuperTokenFactoryMock.sol";
+import {
+    SuperTokenV1Library
+} from "../../../contracts/apps/SuperTokenV1Library.sol";
+import { ForkBaselineTest } from "./ForkBaseline.t.sol";
+
+/// @title ForkSuperTokenFactoryUpgradeTest
+/// @author Superfluid
+/// @notice Tests the SuperTokenFactory upgrade flow on a forked mainnet
+/// @dev Note that this test file is likely dynamic and will change over time
+/// due to the possibility that the upgrade flow may also change over time
+contract ForkSuperTokenFactoryUpgradeTest is ForkBaselineTest {
+    using SuperTokenV1Library for ISuperToken;
+    string public PROVIDER_URL;
+
+    IResolver public constant resolver =
```

Comment:
It is not very clear from the context that this is for Polygon only.

Reply:
addressed: f22d6aee070fdeba4d2c4d6e0144bb4272dfeefd